lucene/core/src/test/org/apache/lucene/index/TestDocumentsWriterDeleteQueue.java

```diff
@@ -36,6 +37,26 @@
  */
 public class TestDocumentsWriterDeleteQueue extends LuceneTestCase {
 
+
+  public void testAdvanceReferencesOriginal() {
+    WeakAndNext weakAndNext = new WeakAndNext();
+    DocumentsWriterDeleteQueue next = weakAndNext.next;
+    assertNotNull(next);
+    System.gc();
```

Comment:
I checked openjdk sources and they use System.gc() with a similar expectation that it will always try to flush all weak refs before returning, so this seems fine. The test itself could maybe be made more explicit by utilizing an explicit ReferenceQueue rather than asserting the reference has been cleared but this is a matter of taste - if this works, it will in both cases.

Reply:
cool! I think the simpler the better

Reply:
I would also use a ReferenceQueue to do the check. The weak.get() has a bad taste to me.
I did not check that System.gc() tries to cleanup weak refs, but if it does and we can rely on this, I am fine.